### ISSUE TITLE
Svg arc.cornerRadius causing invalid paths with attrTween

### ISSUE THREAD

**Issue by wassx** [mbostock/d3#2183]
I use the partition layout to create a sunburst chart where the arcs have rounded corners. The animation works pretty well (except a flickering of the arc which becomes a full circle), but in the console of safari 7.1 and chrome 39.0 a lots of errors are thrown (NOT in firefox).
The animation works but an error occurs in d3.js:8699 "this.setAttribute(name, f(t));"
The log output of chrome:
Error: Invalid value for <path> attribute d="MNaN,NaNANaN,NaN 0 1,1 NaN,NaNLNaN,NaNANaN,NaN 0 0,1 NaN,NaNZ"

There is an ongoing discussion on stackoverflow: http://stackoverflow.com/questions/27838938/d3-sunburst-diagram-setting-arc-cornerradius-causes-invalid-path-with-arctwee

and a codepen to test made by user VividD: http://codepen.io/VividD/pen/myRWRY

**Comment by mbostock** [comment]
Without a test case (a self-contained example on bl.ocks.org or jsfiddle.net, etc.) to verify this bug, I can’t investigate further. My guess is that one of the arc properties you’re specifying is NaN, so that NaN is being propagated into the resulting path string. I would use a debugger to step through your code and see what values you are passing to the arc generator, and then work backwards from there to see why the property is NaN.

I did notice one bug in your code:

```js
var range = +y.range();
var yr = d3.interpolate(range, [dy ? selectionRadius : innerBtnRadius, radius-5]);
```

Here you’ve defined `range` by coercing a scale’s range (which is typically an array of numbers) to a number. That probably means that `range`’s value is NaN. Then you’re using d3.interpolate to interpolate from NaN to an array of two numbers. That probably means that the interpolator is always returning the latter array, regardless of the value of *t*.

**Comment by wassx** [comment]
Hi, sry to ask again, but maybe you haven't noticed, there is a example of the zoomable sunburst on codepen in the issue: " and a codepen to test made by user VividD: http://codepen.io/VividD/pen/myRWRY "
Is this one helpful for you?

**Comment by mbostock** [comment]
I did miss the link, thank you. And thank you for debugging the divide-by-zero. You can see that the two lines between intersected are coincident, hence the zero denominator. I think your fix probably isn’t what we want, but I haven’t thought through the appropriate fix to svg/arc.js yet.

**Comment by moneytree-doug** [comment]
Nice good to know this is a milestone already! Thanks!

**Comment by geofflancaster** [comment]
Any idea when this will be released? I'm having this issue trying to draw a simple arc with rounded corners

**Comment by mbostock** [comment]
Fix merged into #2591.
